packages/babel-generator/src/buffer.ts

```diff
@@ -262,11 +242,7 @@ export default class Buffer {
    * over "();", where previously it would have been a single mapping.
    */
   exactSource(loc: any, cb: () => void) {
-    // In cases where parent expressions start at the same locations as the
-    // identifier itself, the current active location could already be the
-    // start of this range. We use 'force' here to explicitly start a new
-    // mapping range for this new token.
-    this.source("start", loc, true /* force */);
+    this.source("start", loc);
```

Comment:
We no longer need to force. `gen-mapping` can determine that a name index has changed, and will preserve the mapping.

Eg, `const foo = x => y;`, we have two mappings pointing at the `x` variable. The first mapping is the arrow function, and it does not have a `name` field. The second mapping is for the identifier, and it does have a `name`. Because the name changed between the mappings, we'll keep both mappings instead of skipping the second. This is what `force` accomplished because we didn't check for `identifierName` changes in the `lastXyz === xyz` section.